### DISCUSSION TITLE
hinge triggers immediately

### DISCUSSION THREAD

**Discussion by Cyberchicken** [animate-css/animate.css#1466] [Q&A] [answered]
Applying "animate__hinge" class to an element triggers the animation as soon as the page is loaded, 
while the documentation suggests that the trigger should be the "animate__animated" class.

The demo works beacause it adds both classes at once.

My setup to reproduce on Google Chrome Version 97.0.4692.71 (Official Build) (64-bit):
```
<!doctype html><head><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/animate.css/4.1.1/animate.min.css"/></head>
<div class="animate__hinge"><img src="https://picsum.photos/300/200"></div>
```

Expected behavior: no animation.

The few other animations I tried worked as I expected.
Did I get something wrong?

I will experiment with Animate.css-Dynamic, with using the bare keyframes definitions and by slicing down Animate.css to the minimum components.

Thank you

**Comment by eltonmesquita** [comment]
See https://github.com/animate-css/animate.css/discussions/1453

**Comment by Cyberchicken** [comment]
I think I deserve an answer: if "animate__animated" is not the trigger, what's its purpose?

**Comment by eltonmesquita** [comment, marked as answer]
`animate__animated` is just a helper, it doesn't "hold" the animation keyframe, it just gets some animation related properties, as you [can see here](https://github.com/animate-css/animate.css/blob/main/source/_base.css):
```css
.animated {
  animation-duration: var(--animate-duration);
  animation-fill-mode: both;
}
```
If you don't want the animation to be triggered at page load, it's better to add both classes with js, not only one. You can also just set the CSS `animation` property yourself and not use the helper class

**Reply by Cyberchicken** [reply]
I see, it's just a light factoring.

> If you don't want the animation to be triggered at page load, it's better to add both classes with js, not only one. You can also just set the CSS animation property yourself and not use the helper class

In a backend like wordpress's you just have access to additional classes, not data. Unless you develop a proper editor.

I'll try to reshuffle animate.css to be used in such environment: adding classes like "animate_onload", "animate_ondisplay" etc
Thank you.

**Comment by Cyberchicken** [comment]
The "problem" with Hinge is that it's one of the few animations that have a redefinition of animation-duration (the others beiing heartBeat, bounceIn/Out, flipOutX/Y).
Therefore animate_animated practically acts as a gate for all other animations.
